ma/config/magento.py

```diff
@@ -11,3 +11,40 @@
 
 _LOGGER.debug("CONFIG: S=[%s] U=[%s] P=[%s] H=[%s] P=(%d)",
               SCHEME, USERNAME, PASSWORD, HOSTNAME, PORT)
+
+# Hash used to execute remote re-indexing and cache refresh.
+UTILITY_HASH = os.environ.get('MAGENTO_UTILITY_HASH', None)
```

Comment:
Hmm. Why wouldn't you want a cache-problem to interrupt execution? In most
scalable applications (which this is not) caching is a core component of
the model. If the caching doesn't work the whole thing breaks down.

My other company has the same position, but they're also young programmers.
I specifically don't like it when any of the layers aren't in place,
especially when you're using a cache-layer hosted on Amazon (which ours is
not), but you usually have a very solid rationale for your decisions.



Dustin


On Mon, May 16, 2016 at 7:49 PM, Diego Terzano <notifications@github.com>
wrote:

> In ma/config/magento.py
> <https://github.com/CoffeeForThinkers/MagentoAPI/pull/3#discussion_r63445542>
> :
>
> > @@ -11,3 +11,40 @@
> >
> >  _LOGGER.debug("CONFIG: S=[%s] U=[%s] P=[%s] H=[%s] P=(%d)",
> >                SCHEME, USERNAME, PASSWORD, HOSTNAME, PORT)
> > +
> > +# Hash used to execute remote re-indexing and cache refresh.
> > +UTILITY_HASH = os.environ.get('MAGENTO_UTILITY_HASH', None)
>
> I renamed it to MAGENTO_SECURITY_TOKEN.
> I don't want to throw an exception if the variable is not set.
> Caching/Indexing should not prevent the process to run. I modified to raise
> the exception upon class instantiation, if missing. at least the main
> functionality would run by then.
> I initially added the exception to both methods because I envisioned using
> MaintenanceApi for other things that might not require the Security Token.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/CoffeeForThinkers/MagentoAPI/pull/3/files/d7d16fde3fcbb5a5961df6774e6a02f5e1ceb5ea..13e52d77a91a81728accfd069105c16df56f6010#r63445542>
>